File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    int answer;
    
    public int solution(int[] numbers, int target) {        
        DFS(0, 0, numbers, target);
        return answer;
    }
    
    public void DFS(int index, int value, int[] numbers, int target){
        if(index >= numbers.length) {
            if(value == target){
                answer++;
            }
            return;
        }
        
        DFS(index+1, value+numbers[index], numbers, target);
        DFS(index+1, value-numbers[index], numbers, target);

    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    n = len(numbers)
    
    
    def dfs(result, index):
        nonlocal answer
        
        if index == n and result == target:
            answer +=1
            
        if index < n:
            dfs(result+numbers[index], index +1)
            dfs(result-numbers[index], index +1)
    
    dfs(0, 0)
    
    return answer